### PR TITLE
[Layout foundations] Replace Stack with AlphaStack

### LINKED ISSUE #8102: [Layout foundations] [Beta] Replace `Stack` with `AlphaStack`

### PR DESCRIPTION
**Pull request by laurkim**
### WHY are these changes introduced?

Resolves #8102.

Replaces deprecated `Stack` component with `AlphaStack`.

### WHAT is this pull request doing?

- Replaces deprecated Stack with AlphaStack in `polaris-react`
- Updates examples and components using AlphaStack to use Stack
- Removes deprecated Stack documentation and examples
- Updates AlphaStack documentation and examples

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris.shopify.com/next.config.js [hunk 1]
```diff
@@ -346,8 +341,8 @@ const layoutAndStructure = [
     permanent: false,
   },
   {
-    source: '/components/alpha-stack',
-    destination: '/components/layout-and-structure/alpha-stack',
+    source: '/components/stack',
+    destination: '/components/layout-and-structure/stack',
```
Comment (kyledurand) [inline polaris.shopify.com/next.config.js]: I think we'll need to add a
```js
source: '/components/layout-and-structure/alpha-stack'
destination: '/components/layout-and-structure/stack'
```

**Review by kyledurand — APPROVED** [review]
So glad we're removing this prefix. Just a comment around adding a redirect. We might have some links out there for alpha-stack and we want to point those to stack now

**Review by aveline — APPROVED** [review]
Just want to note that if this gets merged in before https://github.com/Shopify/polaris/pull/8582 and https://github.com/Shopify/polaris/pull/8588 those PRs will have to be updated accordingly.

**Comment by laurkim** [comment]
> Just want to note that if this gets merged in before #8582 and #8588 those PRs will have to be updated accordingly.

@aveline I'll update the v11 branch when those go in 👍

**Pull request merged**